controlplane/kubeadm/controllers/scale.go

```diff
@@ -90,13 +82,16 @@ func (r *KubeadmControlPlaneReconciler) scaleDownControlPlane(
 		return ctrl.Result{}, errors.Wrapf(err, "failed to create client to workload cluster")
 	}
 
-	// We don't want to health check at the beginning of this method to avoid blocking re-entrancy
-
 	// Wait for any delete in progress to complete before deleting another Machine
 	if controlPlane.HasDeletingMachine() {
 		return ctrl.Result{}, &capierrors.RequeueAfterError{RequeueAfter: deleteRequeueAfter}
 	}
 
+	// We don't want to health check at the beginning of this method to avoid blocking re-entrancy
```

Comment:
@detiber you added this in #2381 but I'm not sure I understand what the problem with healthchecking and re-entrancy is/was. Can you comment?

Reply:
We ran into a few cases where the etcd health check was failing because there was one fewer etcd member than there were control plane Machines – checking at the top would have blocked the KCP from taking action to improve the situation on the next reconcile.

I'm not aware of any of those cases that are still present, though (see above for my fumbling attempts to identify one such flow).